### ISSUE TITLE
SFTP and/or FTP support

### ISSUE THREAD

**Issue by klokan** [klokan/webstor#7]
How hard would it be to support sftp / ftp for uploading in this utility?
Vasek, could you please provide estimate of required working hours?
Tip: http://curl.haxx.se/libcurl/c/sftpget.html

**Comment by xrosecky** [comment]
You want to extended webstor to support SFTP/FTP protocol or utility for uploading MBTile using these protocols?

**Comment by klokan** [comment]
Well, yes, possibly. But first I want to know how hard it would be.
I need an estimate with range of hours it would probably take you to do that...

**Comment by xrosecky** [comment]
OK, webstor uses libcurl library, so at least the I/O polling part can be reused. I will take a look at it and give you an estimate.

**Comment by klokan** [comment]
Great, thanks!

**Comment by xrosecky** [comment]
Yes, it can be reused, what operations do you want to implement?

I can implement for a start uploading of MBTiles using SFTP/FTP, it takes a few hours (2 - 6).

**Comment by klokan** [comment]
OK. Please start with the implementation of uploading for SSH and FTP.
6 hours cap sounds OK.

On Thu, Mar 21, 2013 at 8:46 PM, Václav Rosecký notifications@github.comwrote:

> Yes, it can be reused, what operations do you want to implement?
> 
> I can implement for a start uploading of MBTiles using SFTP/FTP, it takes
> a few hours (2 - 6).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/klokan/webstor/issues/7#issuecomment-15261103
> .

## 

Petr Pridal
(managing director)

Klokan Technologies GmbH
Bahnhofstrasse 9, Baar 6340, Switzerland
Tel: +41 (0)41 511 26 12
Email: info@klokantech.com
Web: http://www.klokantech.com/

**Comment by xrosecky** [comment]
I got stuck on it, it hangs when using SFTP protocol with asynchronous requests and I don't know why. I created separate branch for reference with my current state (https://github.com/klokan/webstor/tree/async_sftp).

Synchronous requests (no I/O polling) are OK.

**Comment by klokan** [comment]
What about simple FTP? Are there troubles as well?

If there are serious problems, let's postpone this development for now and start to work on the #9, please.

**Comment by xrosecky** [comment]
Uploads using FTP are working, the problem is with SFTP protocol.
